File: tests/fixtures/golden/typescript/caller.ts
import axios from "axios";
import { run } from "./ambiguous_a";
import { Dog, Speaker } from "./animals";
import { Greeter } from "./greeter";
import { formatGreeting } from "./helper";
import * as helperNs from "./helper";

function localUtil(): string {
  return "local";
}

// Plain call: a bare same-module function call.
export function entry(): string {
  return localUtil();
}

// Named import call: `formatGreeting` is bound by the import only.
export function callImportedHelper(name: string): string {
  return formatGreeting(name);
}

// Namespace import call: `helperNs.formatGreeting`.
export function callNamespaceImport(name: string): string {
  return helperNs.formatGreeting(name);
}

// Receiver-typed call: `g: Greeter` pins `.greet`.
export function callReceiverTyped(g: Greeter): string {
  return g.greet("world");
}

// Inherited method call: `Dog` has no `speak` of its own.
export function callInherited(d: Dog): string {
  return d.speak();
}

// Interface call: dispatch through `Speaker`.
export function callInterface(s: Speaker): string {
  return s.say();
}

// Ambiguous name: `run` exists in two modules; the import picks
// `ambiguous_a`.
export function callAmbiguous(): void {
  run();
}

// External call: `axios` is never indexed.
export function callExternal(): void {
  axios.get("https://example.com");
}

File: tests/fixtures/golden/python/caller.py
import requests

from greeter import Greeter
from animals import Dog
from helper import format_greeting


def local_util() -> str:
    return "local"


def entry() -> str:
    """Plain call: a bare same-module function call."""
    return local_util()


def call_imported_helper(name: str) -> str:
    """`from x import y` call: `format_greeting` is bound by name only,
    not through an attribute access on `helper`."""
    return format_greeting(name)


def call_receiver_typed(g: Greeter) -> str:
    """Receiver-typed call: `g`'s annotation pins its type, so `.greet`
    must resolve to `Greeter.greet` and nothing else."""
    return g.greet("world")


def call_inherited(d: Dog) -> str:
    """Inherited method call: `Dog` has no `speak` of its own — must
    dispatch to `Animal.speak` via `Dog`'s recorded EXTENDS edge."""
    return d.speak()


def call_ambiguous() -> None:
    """Ambiguous bare name: `run` exists in two unrelated modules and is
    imported from neither here — must stay unresolved, not guess."""
    run()


def call_external() -> None:
    """Call into an external library: `requests` is never indexed (it
    is not part of this fixture repo), so this must stay unresolved."""
    requests.get("https://example.com")
